### PR TITLE
Ref #17769 - Replaced superglobal in LintController

### PR DESCRIPTION
**Pull request by Maria-Papa**
Signed-off-by: Maria Papadopoulou <1maria.papa@gmail.com>

### Description

Please describe your pull request: This is a partial fix of the issue #17769.

Fixes #

Before submitting pull request, please review the following checklist:

- [X] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [X] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [X] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [X] Every commit has a descriptive commit message.
- [X] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [X] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]

File: libraries/classes/Controllers/Database/SqlController.php [hunk 1]
```diff
@@ -56,14 +56,15 @@ public function __invoke(ServerRequest $request): void
...
+        $delimiter = $request->getParsedBodyParam('delimiter');
 
         $this->response->addHTML($this->sqlQueryForm->getHtml(
             $GLOBALS['db'],
             '',
             true,
             false,
-            isset($_POST['delimiter'])
-                ? htmlspecialchars($request->getParsedBodyParam('delimiter'))
+            isset($delimiter)
```
Comment (williamdes) [inline libraries/classes/Controllers/Database/SqlController.php:66]:
```suggestion
            $delimiter !== null
```
it is always isset ;)

File: libraries/classes/Controllers/UserPasswordController.php [hunk 2]
```diff
@@ -62,15 +62,18 @@ public function __invoke(ServerRequest $request): void
...
+        $noPass = $request->getParsedBodyParam('nopass');
+        $pmaPw = $request->getParsedBodyParam('pma_pw');
+
         /**
          * If the "change password" form has been submitted, checks for valid values
          * and submit the query or logout
          */
-        if (isset($_POST['nopass'])) {
-            if ($_POST['nopass'] == '1') {
+        if (isset($noPass])) {
```
Comment (williamdes) [inline libraries/classes/Controllers/UserPasswordController.php:72]: same, it is always isset + typo fix
```suggestion
        if ($noPass !== null) {
```

File: libraries/classes/Controllers/Database/SearchController.php [hunk 3]
```diff
@@ -79,7 +79,7 @@ public function __invoke(ServerRequest $request): void
         }
 
         // Main search form has been submitted, get results
-        if (isset($_POST['submit_search'])) {
+        if (isset($request->getParsedBodyParam('submit_search'))) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Database/SearchController.php:82]:
```suggestion
        if ($request->getParsedBodyParam('submit_search') !== null) {
```
It's definitely set. Maybe check for `!== null`?

**Review by kamil-tekiela — CHANGES_REQUESTED** [review]

**Review by kamil-tekiela — APPROVED** [review]
LGTM

**Review by williamdes — APPROVED** [review]

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**